sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -127,6 +131,11 @@ AZ_NODISCARD az_result az_keyvault_keys_client_init(
     }
   };
 
+  if (!az_http_policy_retry_options_validate(&self->_internal.options.retry))
```

Comment:
We don't need this

Reply:
I am not objecting, but at a time I wrote checks in the policy implementation, and you told me that the clients should validate options on creation instead, so that when we are inside the policy, we know that max_attempts, delay and max_delay - all are non-negative, and delay is <= max_delay, so I made this function.

Reply:
Done :)